rest-fights/src/main/java/io/quarkus/sample/superheroes/fight/client/VillainClient.java

```diff
@@ -25,50 +29,120 @@
  */
 @ApplicationScoped
 public class VillainClient {
-	private final WebTarget villainClient;
-
-	public VillainClient(FightConfig fightConfig) {
-		this.villainClient = ClientBuilder.newClient()
-			.target(fightConfig.villain().clientBaseUrl())
-			.path("api/villains/");
-	}
-
-	/**
-	 * Gets a Villain from the Villain service wrapped with a recovery on a {@code 404} error. Also wrapped in a {@link CircuitBreaker}.
-	 * @return The Villain
-	 */
-	@CircuitBreaker(requestVolumeThreshold = 8, failureRatio = 0.5, delay = 2, delayUnit = ChronoUnit.SECONDS)
-	@CircuitBreakerName("findRandomVillain")
-	CompletionStage<Villain> getRandomVillain() {
-		// Want the 404 handling to be part of the circuit breaker
-		// This means that the 404 responses aren't considered errors by the circuit breaker
-		return this.villainClient.path("random")
-			.request(MediaType.APPLICATION_JSON_TYPE)
-			.rx(UniInvoker.class)
-			.get(Villain.class)
-			.onFailure(Is404Exception.IS_404).recoverWithNull()
-			.subscribeAsCompletionStage();
-	}
-
-	/**
-	 * Finds a random {@link Villain}. The retry logic is applied to the result of the {@link CircuitBreaker}, meaning that retries that return failures could trigger the breaker to open.
-	 * @return A random {@link Villain}
-	 */
-	public Uni<Villain> findRandomVillain() {
-		// The CompletionState is important so that on retry the Uni re-subscribes to a new
-		// CompletionStage rather than the original one (which has already completed)
-		return Uni.createFrom().completionStage(this::getRandomVillain)
-			.onFailure().retry().withBackOff(Duration.ofMillis(200)).atMost(3);
-	}
+  private static final String STORK_PREFIX = "stork://";
```

Comment:
I'm not sure at the moment, but I think using `stork://` in the web target should also work (and would keep this code simpler).
Have you tried it?

Reply:
I didn't try it. When I talked with @cescoffier before doing this I believe he mentioned it was only supported when using the rest client.

I will certainly try it though. If it works I can most likely revert all the changes in this class.

Reply:
@michalszynkiewicz is there info/debug logging in Stork that I can turn on to see whats going on under the covers?

Reply:
I haven't done much work on the kube stuff, @aureamunoz knows best.

Re tracing/logging, that's an area we probably should improve in Stork...